src/main/java/org/truffleruby/core/numeric/IntegerNodes.java

```diff
@@ -888,6 +888,30 @@ public int compare(long a, DynamicObject b) {
             return BigInteger.valueOf(a).compareTo(Layouts.BIGNUM.getValue(b));
         }
 
+        @Specialization
+        public int compare(DynamicObject a, long b) {
+            return Layouts.BIGNUM.getValue(a).compareTo(BigInteger.valueOf(b));
+        }
+
+        @Specialization(guards = "!isInfinity(b)")
+        public int compare(DynamicObject a, double b) {
+            return Double.compare(Layouts.BIGNUM.getValue(a).doubleValue(), b);
+        }
+
+        @Specialization(guards = "isInfinity(b)")
+        public int compareInfinity(DynamicObject a, double b) {
+            if (b < 0) {
+                return +1;
```

Comment:
I don't think we do this explicit `+1` in other compare method implementations.

Reply:
It's a nice symmetry, but I can remove if you prefer.

Reply:
I don't feel strongly enough about it. It looks slightly odd to me, but if you like it, feel free to keep it.